content/1_docs/2_cookbook/9_setup/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,468 @@
+
+On the other hand, if you have a `book` object, but try to use a method that is not defined in the class, you will also get an error, this time 
+
+```
+Fatal error: Uncaught Error: Call to undefined method Book::type()
+```
+
+You can create as many book objects from this class as you like, by providing the minimum properties required to define a book (in this case the title and the author).
+
+<info>
+In our example above, we have also introduced type declarations for parameters and return types. This concept is not limited to OOP, but makes sure that we get the correct input values and that our methods return what they are expected to return. An error is thrown when the type declarations are not met.
+</info>
+
+### The constructor
+
+In the example above, we declared a `__constructor()` method. This method is optional, but allows us to initialize an object's properties (or the properties we regard as essential) upon creation of the object. In our example, we have to pass the title and the author parameters both as strings when creating a new instance of the class and cannot just create a new Book with `new Book()`. It is also possible to set default values for certain parameters (which doesn't really make sense for our books). But all of this totally depends on the class and the objectives.
+
+## Classes and objects in Kirby
+
+After this short intro to classes and objects in PHP, we should be prepared to dive a little into Kirby's classes which will also extend our knowledge of OOP that we have covered so far.
+
+## The `Page` class
+
+The `Page` class is one of the – if not *the* – most important classes in Kirby: after all, content in Kirby is based on the concept of pages. A folder in the `/content` folder represents a page.
+
+The `Page` class is defined in `/kirby/src/Cms/Page.php`. In your Starterkit or Plainkit, open this file in an editor and have a look at it.
+
+Looks a bit different than our mini example above, doesn't it? It's not only very much longer but it doesn't even start with the class definition. Let's go through what this stuff at the top means.
+
+### Namespacing
+The first line after the opening PHP tag reads:
+
+```php "/kirby/src/Cms/Page.php"
+namespace Kirby\Cms;
+```
+
+Here, we declare a (link: https://www.php.net/manual/en/language.namespaces.rationale.php text: namespace ) for the class to prevent name collisions with other classes of the same name (for example classes from third party libraries we want to use in our project) and to logically group classes that belong together. 
```

Comment:
I'd explain here that the `Page` class becomes `Kirby\Cms\Page` because of the namespace. That makes the following section easier to understand.